src/analysis/analysis-poller.js

```diff
@@ -1,22 +1,32 @@
 var _ = require('underscore');
 var BackbonePoller = require('backbone-poller');
 
-var AnalysisPoller = function () {};
+var AnalysisPoller = function () {
+  this._pollers = [];
+};
 
 AnalysisPoller.CONFIG = {
   START_DELAY: 1000,
   MAX_DELAY: Infinity,
   DELAY_MULTIPLIER: 1.5
 };
 
-AnalysisPoller.prototype.poll = function (analysisModel) {
-  this._pollers = [];
-  var poller = this._findOrCreatePoller(analysisModel);
-  if (analysisModel.hasChanged('url') && poller.active()) {
-    poller.stop();
-  }
-  if (!analysisModel.isDone()) {
-    poller.start();
+AnalysisPoller.prototype.resetAnalysisNodes = function (analysisModels) {
+  this.reset();
+  _.each(analysisModels, function (analysisModel) {
+    this._poll(analysisModel);
+  }, this);
+};
+
+AnalysisPoller.prototype._poll = function (analysisModel) {
+  if (analysisModel.url()) {
```

Comment:
What is the semantic meaning of this?

An analysis with an url is considered as ready?

Reply:
Analyses don't have a `url` initially. Only after the first request to Maps API they are given a `url`. That's when polling starts so the truth is that every analysisModel should have a `url` at this point. Polling is only possible if there's a `url` and I guess this check is just ensuring polling won't start unless that's true. It's possible that this could work with this check, but it's also not harmful...

Reply:
In this case maybe we should create a function for this conditional.
([something like this](https://sourcemaking.com/refactoring/decompose-conditional))

```javascript
AnalysisPoller.prototype._poll = function (analysisModel) {
  if (this._canBePolled(analysisModel)) {
    // ... 
  }
}

/**
 * We need the analysis to have an url in order to start polling
 */
AnalysisPoller.prototype._canBePolled = function (analysisModel) {
  return !!analysisModel.url();
}

Reply:
I dont understand the intetion of this:

```javascript
if (analysisModel.hasChanged('url') && poller.active()) {
      poller.stop();
    }
    if (!analysisModel.isDone()) {
      poller.start();
    }
```